range panics if not used in an assign expression

The following program `sample.go` triggers a panic:

```go
package main

func main() {
        for range []struct{}{} {
        }
}       
```

Expected result:
```console
$ go run ./sample.go

```

Got:
```console
$ yaegi ./sample.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x618640]

goroutine 1 [running]:
github.com/containous/yaegi/interp.(*Interpreter).cfg.func1(0xc0000daf00, 0xc000185d68)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/cfg.go:70 +0x1220
github.com/containous/yaegi/interp.(*node).Walk(0xc0000daf00, 0xc000185d68, 0xc000185da0)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/interp.go:106 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc0000dab00, 0xc000185d68, 0xc000185da0)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000daa00, 0xc000185d68, 0xc000185da0)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000da900, 0xc000185d68, 0xc000185da0)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000da400, 0xc000185d68, 0xc000185da0)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000da200, 0xc000185d68, 0xc000185da0)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc0000aa540, 0xc0000da200, 0xc0000a4650, 0x4, 0x0, 0x0, 0x4)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/cfg.go:39 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000aa540, 0xc0000c00c0, 0x39, 0x239, 0xc0000c00c0, 0x39, 0x0, 0xc2f260)
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/interp/interp.go:257 +0x135
main.main()
        /home/user/go/pkg/mod/github.com/containous/yaegi@v0.0.1/cmd/yaegi/yaegi.go:77 +0x2e3
```

--- a/_test/a30.go
+++ b/_test/a30.go
@@ -1,0 +1,10 @@
+package main
+
+func main() {
+	for range []struct{}{} {
+	}
+	println("ok")
+}
+
+// Output:
+// ok

--- a/_test/a31.go
+++ b/_test/a31.go
@@ -1,0 +1,11 @@
+package main
+
+func main() {
+	for range []int{0, 1, 2} {
+		print("hello ")
+	}
+	println("")
+}
+
+// Output:
+// hello hello hello

--- a/interp/ast.go
+++ b/interp/ast.go
@@ -696,7 +696,13 @@ func (interp *Interpreter) ast(src, name string) (string, *node, error) {
 		case *ast.RangeStmt:
 			// Insert a missing ForRangeStmt for AST correctness
 			n := addChild(&root, anc, pos, forRangeStmt, aNop)
-			st.push(addChild(&root, astNode{n, nod}, pos, rangeStmt, aRange), nod)
+			r := addChild(&root, astNode{n, nod}, pos, rangeStmt, aRange)
+			st.push(r, nod)
+			if a.Key == nil {
+				// range not in an assign expression: insert a "_" key variable to store iteration index
+				k := addChild(&root, astNode{r, nod}, pos, identExpr, aNop)
+				k.ident = "_"
+			}
 
 		case *ast.ReturnStmt:
 			st.push(addChild(&root, anc, pos, returnStmt, aReturn), nod)

